### ISSUE TITLE
tests segfault on OS X

### ISSUE THREAD

**Issue by ongardie-ebay** [rust-rocksdb/rust-rocksdb#168]
I ran this on f784fc26 on macOS Sierra (10.12.6).

```
~/rust-rocksdb:master$ cargo test -vvv
[...snip...]
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `/Users/dongaro/rust-rocksdb/target/debug/deps/rocksdb-c4d44a0ec3b28825`

running 10 tests
test db::errors_do_stuff ... ok
test db::external ... ok
test compaction_filter::compaction_filter_test ... ok
test db::iterator_test ... ok
test db::test_db_vector ... ok
test db_options::tests::test_enable_statistics ... ok
test db::snapshot_test ... ok
test db::writebatch_works ... ok
test merge_operator::test::mergetest ... ok
test merge_operator::test::counting_mergetest ... ok

test result: ok. 10 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running `/Users/dongaro/rust-rocksdb/target/debug/deps/test_column_family-171e54ab4ded8d69`

running 4 tests
test test_merge_operator ... ignored
test test_create_missing_column_family ... ok
test test_column_family_with_options ... ok
test test_column_family ... ok

test result: ok. 3 passed; 0 failed; 1 ignored; 0 measured; 0 filtered out

error: process didn't exit successfully: `/Users/dongaro/rust-rocksdb/target/debug/deps/test_column_family-171e54ab4ded8d69` (signal: 11, SIGSEGV: invalid memory reference)
~/rust-rocksdb:master$ lldb  /Users/dongaro/rust-rocksdb/target/debug/deps/test_column_family-171e54ab4ded8d69
(lldb) target create "/Users/dongaro/rust-rocksdb/target/debug/deps/test_column_family-171e54ab4ded8d69"
Current executable set to '/Users/dongaro/rust-rocksdb/target/debug/deps/test_column_family-171e54ab4ded8d69' (x86_64).
(lldb) r
Process 71092 launched: '/Users/dongaro/rust-rocksdb/target/debug/deps/test_column_family-171e54ab4ded8d69' (x86_64)

running 4 tests
test test_merge_operator ... ignored
test test_create_missing_column_family ... ok
test test_column_family_with_options ... ok
test test_column_family ... ok

test result: ok. 3 passed; 0 failed; 1 ignored; 0 measured; 0 filtered out

Process 71092 stopped
* thread #2, name = 'test_column_family', stop reason = EXC_BAD_ACCESS (code=2, address=0x7fffb59714c4)
    frame #0: 0x00007fffccbefca9 libsystem_platform.dylib`OSSpinLockLock + 7
libsystem_platform.dylib`OSSpinLockLock:
->  0x7fffccbefca9 <+7>:  lock
    0x7fffccbefcaa <+8>:  cmpxchgl %ecx, (%rdi)
    0x7fffccbefcad <+11>: jne    0x7fffccbefcb0            ; <+14>
    0x7fffccbefcaf <+13>: retq
Target 0: (test_column_family-171e54ab4ded8d69) stopped.
(lldb) bt
warning: could not execute support code to read Objective-C class data in the process. This may reduce the quality of type information available.
* thread #2, name = 'test_column_family', stop reason = EXC_BAD_ACCESS (code=2, address=0x7fffb59714c4)
  * frame #0: 0x00007fffccbefca9 libsystem_platform.dylib`OSSpinLockLock + 7
    frame #1: 0x0000000100d183f0 test_column_family-171e54ab4ded8d69`je_tcache_bin_flush_small [inlined] je_malloc_mutex_lock at mutex.h:99 [opt]
    frame #2: 0x0000000100d183db test_column_family-171e54ab4ded8d69`je_tcache_bin_flush_small(tsd=0x0000000102a6b188, tcache=<unavailable>, tbin=0x0000000102ad1028, binind=0, rem=0) at tcache.c:119 [opt]
    frame #3: 0x0000000100d19594 test_column_family-171e54ab4ded8d69`tcache_destroy(tsd=0x0000000102a6b188, tcache=0x0000000102ad1000) at tcache.c:371 [opt]
    frame #4: 0x0000000100d193ca test_column_family-171e54ab4ded8d69`je_tcache_cleanup(tsd=0x0000000102a6b188) at tcache.c:410 [opt]
    frame #5: 0x0000000100d19d00 test_column_family-171e54ab4ded8d69`je_tsd_cleanup(arg=0x0000000102a6b188) at tsd.c:82 [opt]
    frame #6: 0x0000000100d1a8ca test_column_family-171e54ab4ded8d69`je_tsd_cleanup_wrapper(arg=0x0000000102a6b180) at tsd.h:688 [opt]
    frame #7: 0x00007fffccbfc355 libsystem_pthread.dylib`_pthread_tsd_cleanup + 470
    frame #8: 0x00007fffccbfc0d9 libsystem_pthread.dylib`_pthread_exit + 152
    frame #9: 0x00007fffccbfa946 libsystem_pthread.dylib`_pthread_body + 191
    frame #10: 0x00007fffccbfa887 libsystem_pthread.dylib`_pthread_start + 286
    frame #11: 0x00007fffccbfa08d libsystem_pthread.dylib`thread_start + 13
```

**Comment by gagarin55** [comment]
I have same results

**Comment by iSynaptic** [comment]
I think this was fixed with #199.  Can you confirm?

**Comment by iSynaptic** [comment]
#200 shows that tests are running for OS X now. Closing.
